tests/Resources/Page/LoginPage.robot

```diff
@@ -43,7 +51,7 @@ Login To Openshift
     Click Element  xpath=/html/body/div/div/main/div/form/div[4]/button
     # FIXME: replace this sleep for something more efficient, considering that this method is used for
     # authentication in OpenShift Console, but also RHODS dashboard and other places
-    Sleep  10
+    Sleep  2s
```

Comment:
what is the goal of this `Sleep 2s`, what it is supposed to wait for?
couldn''t it be replaced with a `Wait Until Page Contains  <something>` ?

Reply:
Unfortunately this keyword is called to log into Openshift, but after logging in you can be taken to any number of different web pages (e.g. the cluster web console, the rhods dashboard or the JL UI).
I don't think there's a single element we can use to understand when the login is complete, having a list of elements for every possible page we end up in sounds impossible, and waiting until something on the login page disappears doesn't guarantee that the next page is loaded.
As of right now I don't think we can do anything about this, the smallest sleep we can get away with is probably the best option

Reply:
but what's the point of the `Sleep  2s`? 
I mean, I get it clear from your comment that you cannot know what's coming next, but not knowing for not knowing, why waiting?

Reply:
> we try to hand off from one keyword to the next when the keyword is done doing its thing, 

the way I see it (in the few pages I coded) is that a keyword should do what's necessary to move to the next step ... but don't do anything related to the next step,

eg 1, the notebook spawner should let the page "loading" towards the next step (the authentication page, or jupyterlab)
eg 2, the authentication keyword should let the page "loading" towards the next step (the console, the dashboard, jupyterlab)

so `Maybe Skip Tour` doesn't belong to the Authentication keyword, but to the Console page that called it

Reply:
@lugi0 What about if we remove hard coded  sleep and add below code 
`Run Keyword And Ignore Error    Wait Until Keyword Succeeds    retry=2    retry_interval=1sec    Maybe Skip Tour`

Reply:
@tarukumar I think we can actually remove the Sleep now that `Maybe Skip Tour` has an additional check. I'll look into it.

Reply:
Removed the sleep, now `Maybe Skip Tour` uses `Wait Until Page Contains Element` rather than `Page Should Contain` since we can wait out the timeout when we know for sure we are in the ocp web console page.